src/crypto/x509/x509.go

```diff
@@ -1787,7 +1787,7 @@ func buildCertExtensions(template *Certificate, subjectIsEmpty bool, authorityKe
 	}
 
 	// This extension is not critical
-	if template.KeyUsage == KeyUsageDigitalSignature && template.AllowDC && !template.IsCA && !oidInExtensions(oidExtensionDelegatedCredential, template.ExtraExtensions) {
+	if template.AllowDC && !template.IsCA && !oidInExtensions(oidExtensionDelegatedCredential, template.ExtraExtensions) && template.KeyUsage&KeyUsageDigitalSignature != 0 {
```

Comment:
Perhaps add some parentheses here to improve clarity for the bit operation? `...&& (template.KeyUsage & KeyUsageDigitalSignature != 0) {`

Reply:
Added now.